src/app/shared/services/openWeatherMaps.js

```diff
@@ -0,0 +1,42 @@
+export default class OpenWeatherMaps {
+    constructor($http, fiveDayForecast) {
+        this.$http = $http;
+        this.fiveDayForecast = fiveDayForecast;
+    }
+
+    getActualForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/weather';
+
+        const request = {
+            method: 'GET',
+            url: URL,
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        return this.fiveDayForecast = this.$http(request);
+    };
+
+    getFiveDayForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/forecast';
+
+        const request = {
+            method: 'GET',
+            url: URL,
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        return this.fiveDayForecast = this.$http(request);
+    }
```

Comment:
In this functions, do you want to return the response of the request? If that, you are failing :)

Reply:
No, Im returning the promise

Reply:
I am with Jorge, you can return `this.$http.get(URL, params);` or first assign the value to `this.fiveDayForecast` and then return it, but not both.
Additionally, on the second option, `this.fiveDayForecast` will contain the promise, not the value, a better approach can be
````
return this.$http.get(URL, params).then(data => {
    this.fiveDayForecast = data; // maybe data need some transformation before assign it...
});

Reply:
what about async / await? :)